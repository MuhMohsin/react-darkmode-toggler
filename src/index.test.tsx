import React from 'react'
import { shallow } from 'enzyme'
import { DarkModeToggler } from '.'

const clickFn = jest.fn()
describe('ExampleComponent', () => {
  it('general', () => {
    const wrapper = shallow(
      <DarkModeToggler isDark='light' size='small' onClick={clickFn} />
    )
    wrapper.find('div[className="toggle small"]').simulate('click')
    // wrapper.find('div[className="toogle"]').get(1).simulate('click')
    expect(wrapper).toMatchSnapshot()
  })

  test('should return border and small', () => {
    const wrapper = shallow(
      <DarkModeToggler size='small' onClick={jest.fn()} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('should return border value ', () => {
    const wrapper = shallow(
      <DarkModeToggler border='#ffff' onClick={jest.fn()} />
    )
    wrapper.find('div[className="toggle"]').simulate('click')
    expect(wrapper).toMatchSnapshot()
  })
})
