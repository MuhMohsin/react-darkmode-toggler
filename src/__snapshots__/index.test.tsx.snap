// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent general 1`] = `
<div
  className="toggle small"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  style={
    Object {
      "borderColor": "",
    }
  }
>
  <div
    className="toggle__slide"
  />
  <span
    className="toggle__sunlight"
  >
    <FaSun />
  </span>
  <span
    className="toggle__moon"
  >
    <BsMoon />
  </span>
</div>
`;

exports[`ExampleComponent should return border and small 1`] = `
<div
  className="toggle small"
  onClick={[MockFunction]}
  style={
    Object {
      "borderColor": "",
    }
  }
>
  <div
    className="toggle__slide"
  />
  <span
    className="toggle__sunlight"
  >
    <FaSun />
  </span>
  <span
    className="toggle__moon"
  >
    <BsMoon />
  </span>
</div>
`;

exports[`ExampleComponent should return border value  1`] = `
<div
  className="toggle"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  style={
    Object {
      "borderColor": "",
    }
  }
>
  <div
    className="toggle__slide"
  />
  <span
    className="toggle__sunlight"
  >
    <FaSun />
  </span>
  <span
    className="toggle__moon"
  >
    <BsMoon />
  </span>
</div>
`;
